tests/foreman/api/test_hostcollection.py

```diff
@@ -15,446 +15,452 @@
-                req = entities.HostSubscription(host=host_id).subscriptions()
-                assert (
-                    prod_name in req['results'][0]['product_name']
-                ), 'Subscription not applied to HC members'
-            # Remove the subscription
-            # Call nailgun to make the API PUT to members of Host Collection
-            entities.Host().bulk_remove_subscriptions(
-                data={
-                    "organization_id": self.org.id,
-                    "included": {"ids": host_ids},
-                    "subscriptions": [{"id": product_subscription.id, "quantity": 1}],
-                }
-            )
-            # GET the subscriptions from hosts and assert they are gone
-            for host_id in host_ids:
-                req = entities.HostSubscription(host=host_id).subscriptions()
-                assert not req['results'], 'Subscription not removed from HC members'
+
+@pytest.mark.tier1
+def test_positive_update_unlimited_hosts(module_org):
+    """Check if host collection 'unlimited hosts' parameter can be updated
+
+    :id: 09a3973d-9832-4255-87bf-f9eaeab4aee8
+
+    :expectedresults: Host collection 'unlimited hosts' parameter was
+        updated
+
+    :CaseImportance: Critical
+    """
+    random_unlimited = choice([True, False])
+    host_collection = entities.HostCollection(
+        max_hosts=1 if not random_unlimited else None,
+        organization=module_org,
+        unlimited_hosts=random_unlimited,
+    ).create()
+    for unlimited in (not random_unlimited, random_unlimited):
+        host_collection.max_hosts = 1 if not unlimited else None
+        host_collection.unlimited_hosts = unlimited
+        host_collection = host_collection.update(['max_hosts', 'unlimited_hosts'])
+        assert host_collection.unlimited_hosts == unlimited
+
+
+@pytest.mark.tier1
+def test_positive_update_host(module_org, fake_hosts):
+    """Update host collection's host.
+
+    :id: 23082854-abcf-4085-be9c-a5d155446acb
+
+    :expectedresults: The host collection was updated with a new host.
+
+    :CaseImportance: Critical
+    """
+    host_collection = entities.HostCollection(
+        host=[fake_hosts[0]], organization=module_org
+    ).create()
+    host_collection.host_ids = [fake_hosts[1].id]
+    host_collection = host_collection.update(['host_ids'])
+    assert host_collection.host[0].id == fake_hosts[1].id
+
+
+@pytest.mark.upgrade
+@pytest.mark.tier1
+def test_positive_update_hosts(module_org, fake_hosts):
+    """Update host collection's hosts.
+
+    :id: 0433b37d-ae16-456f-a51d-c7b800334861
+
+    :expectedresults: The host collection was updated with new hosts.
+
+    :CaseImportance: Critical
+    """
+    host_collection = entities.HostCollection(host=fake_hosts, organization=module_org).create()
+    new_hosts = [entities.Host(organization=module_org).create() for _ in range(2)]
+    host_ids = f"""[{','.join([f"{host.id}" for host in new_hosts])}]"""
```

Comment:
Is this also a place where you can simplify to `{str(host.id) for host in fake_hosts}`?

If not I think it would be very helpful to add some inline comments here describing the formatted string and list that you're trying to parse through between these two lines.

Reply:
@mshriver sorry I missed that one, I have fixed it now

Thank you